Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.cpp

```diff
@@ -97,27 +97,16 @@ namespace AZ
         // Helper function to get a file reference and create a corresponding job dependency
         bool AddDependency(FindPassReferenceAssetParams& params, AssetBuilderSDK::JobDescriptor* job)
         {
-            AZStd::string_view& file = params.dependencySourceFile;
-            AZ::Data::AssetInfo sourceInfo;
-            AZStd::string watchFolder;
-            bool fileFound = false;
-            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(fileFound, &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath, file.data(), sourceInfo, watchFolder);
-
-            if (fileFound)
```

Comment:
You don't need to handle the case if the file isn't found?

Reply:
No you shouldn't need it. The AP accepts dependencies on files that don't exist but might exist in the future, and will reprocess the dependent jobs if/when those files appear later. This was verified when I tested: "deleting a .shader file with OrderOnce dependency did trigger the .pass file to rebuild and fail." And when I restored the .shader file the .pass file was reprocessed and passed.